src/app/wallets/create-on-chain-address.ts

```diff
@@ -34,28 +37,70 @@ const createOnChainAddress = async (
   return savedOnChainAddress.address
 }
 
-export const createOnChainAddressByWallet = async (
-  wallet: Wallet,
-): Promise<OnChainAddress | ApplicationError> => {
+const createOnChainAddress = async ({
+  walletId,
+  requestId,
+}: {
+  walletId: WalletId
+  requestId?: OnChainAddressRequestId
+}) => {
+  const wallet = await WalletsRepository().findById(walletId)
+  if (wallet instanceof Error) return wallet
+
+  const limitOk = await checkOnChainAddressAccountIdLimits(wallet.accountId)
+  if (limitOk instanceof Error) return limitOk
+
+  const onChainAddress = await NewOnChainService().createOnChainAddress(requestId)
+  if (onChainAddress instanceof Error) return onChainAddress
+
+  const onChainAddressesRepo = WalletOnChainAddressesRepository()
+  const savedOnChainAddress = await onChainAddressesRepo.persistNew({
+    walletId,
+    onChainAddress,
+  })
+  if (savedOnChainAddress instanceof Error) return savedOnChainAddress
+
+  return savedOnChainAddress.address
+}
+
+export const createOnChainAddressByWallet = async ({
+  wallet,
+  requestId,
+}: {
+  wallet: Wallet
+  requestId?: OnChainAddressRequestId
+}): Promise<OnChainAddress | ApplicationError> => {
   if (wallet.currency === WalletCurrency.Btc) {
-    return createOnChainAddressForBtcWallet(wallet.id)
+    return createOnChainAddressForBtcWallet({ walletId: wallet.id, requestId })
   }
 
-  return createOnChainAddressForUsdWallet(wallet.id)
+  return createOnChainAddressForUsdWallet({ walletId: wallet.id, requestId })
 }
 
-export const createOnChainAddressForBtcWallet = async (
-  walletId: WalletId,
-): Promise<OnChainAddress | ApplicationError> => {
+export const createOnChainAddressForBtcWallet = async ({
+  walletId,
+  requestId,
+}: {
+  walletId: WalletId
+  requestId?: OnChainAddressRequestId
+}): Promise<OnChainAddress | ApplicationError> => {
   const validated = await validateIsBtcWallet(walletId)
-  return validated instanceof Error ? validated : createOnChainAddress(walletId)
+  return validated instanceof Error
+    ? validated
+    : createOnChainAddress({ walletId, requestId })
 }
 
-export const createOnChainAddressForUsdWallet = async (
-  walletId: WalletId,
-): Promise<OnChainAddress | ApplicationError> => {
+export const createOnChainAddressForUsdWallet = async ({
```

Comment:
I think it's used [here](https://github.com/GaloyMoney/galoy/blob/c5d1e53249883f2be8fa1a3d4aa40ac32cfcef45/src/app/wallets/create-on-chain-address.ts#L77) which in turn gets used in `add-settled-transaction` and `update-on-chain-receipt`. 

Hm, for payouts it's because of the `amount` argument but I can't remember why for this one. I can double-check if we validated wallet currency for a reason here, and if not maybe we can remove the separation (but maybe in a separate PR since that was in here before).

Reply:
but what is the difference if we enable it for stablesats?... is just the creation of an onchain address

Reply:
It's just that we haven't done it as yet and I don't think we should do that changeset here in addition to introducing bria

Reply:
Agreed we'll enable stablesats onchain receives after this PR is rolled out